Add private channel obfuscation docs (#8454)
discord/discord-api-docs: developers/events/gateway-events.mdx

| large_threshold? | integer                                                                     | Value between 50 and 250, total number of members where the gateway will stop sending offline members in the guild member list      | 50      |
| shard?           | array of two integers (shard_id, num_shards)                                | Used for [Guild Sharding](/developers/events/gateway#sharding)                                                                      | -       |
| presence?        | [update presence](/developers/events/gateway-events#update-presence) object | Presence structure for initial presence information                                                                                 | -       |
| intents          | integer                                                                     | [Gateway Intents](/developers/events/gateway#gateway-intents) you wish to receive                                                   | -       |

> | capabilities?    | integer                                                                     | Opt-in bitfield representing [capabilities](/developers/events/gateway-events#identify-gateway-capabilities) of your gateway client | 0       |

Reviewer: What is the intent behind "opt-in" here?

Author: Reading it now I think it's not necessary. I was initially thinking of this from the perspective of opting in to get this behavior to test with the obfuscation flag, but it's possible we use this for other things in the future so I'll drop it.

<ManualAnchor id="identify-gateway-capabilities" />
###### Gateway Capabilities

> `capabilities` is a bitfield you can use to opt an app into gateway behaviors. It's a separate Identify bitfield from [`intents`](/developers/events/gateway#gateway-intents): `intents` control which events your app receives, while `capabilities` change how existing gateway behaviors work.

Reviewer: ```suggestion
`capabilities` is a bitfield you can use to opt a bot client into gateway behaviors. It's a separate Identify bitfield from [`intents`](/developers/events/gateway#gateway-intents): `intents` control which events your bot client receives, while `capabilities` affects gateway behaviors.
```

| shard?           | array of two integers (shard_id, num_shards)                                | Used for [Guild Sharding](/developers/events/gateway#sharding)                                                                 | -       |
| presence?        | [update presence](/developers/events/gateway-events#update-presence) object | Presence structure for initial presence information                                                                            | -       |
| intents          | integer                                                                     | [Gateway Intents](/developers/events/gateway#gateway-intents) you wish to receive                                              | -       |

> | capabilities?    | integer                                                                     | [gateway capabilities](/developers/events/gateway-events#identify-gateway-capabilities) opt-in bitfield for gateway behaviors  | 0       |

Reviewer: I'd frame this more like "bitfield representing capabilities of your gateway client"

<ManualAnchor id="identify-gateway-capabilities" />
###### Gateway Capabilities

> `capabilities` is a bitfield you can use to opt an app into experimental or in-progress gateway behaviors. It's a separate Identify bitfield from [`intents`](/developers/events/gateway#gateway-intents): `intents` control which events your app receives, while `capabilities` change how existing gateway behaviors work.

Reviewer: same comment here. I'd remove the parts about "experimental or in-progress" - capabilities could be used for anything really so lets not get overly specific. the important part is that its your client indicating that its capable of something

| Flag                | Value     | Description                                                                                                                                                             |
|---------------------|-----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | CHANNEL_OBFUSCATION | `1 << 15` | Opts the app into receiving [obfuscated channel metadata](/developers/resources/channel#channel-object-obfuscated-channels) over the Gateway for channels it can't view |

Reviewer: ```suggestion
| CHANNEL_OBFUSCATION | `1 << 15` | Opts the bot client into receiving [obfuscated channel metadata](/developers/resources/channel#channel-object-obfuscated-channels) over the Gateway for channels it can't view |
```

<Warning>

> `CHANNEL_OBFUSCATION` is a temporary, testing-only opt-in for channel obfuscation. This opt-in mechanism will change before the feature reaches general availability. Obfuscation is then planned to apply to all apps automatically, without requiring this capability. Don't add support for this capability to libraries.

Reviewer: ```suggestion
`CHANNEL_OBFUSCATION` is a temporary, testing-only opt-in for channel obfuscation. This opt-in mechanism will change before the feature reaches general availability. Obfuscation is then planned to apply to all bots automatically, even when they don't provide this capability.
```

I slightly changed my mind, I think its "ok" for a library to implement general capability support, though I agree its not good to implement this particular capability since its temporary. I do wonder if the original language was too strong? Open to other ideas here though

Author: Yeah I think it makes sense to add to libraries with the understanding that this one is temporary. I think softening the wording like you've got here makes sense
